src/snowflake/cli/plugins/streamlit/commands.py

```diff
@@ -134,37 +139,51 @@ def streamlit_deploy(
     stage is used. If the specified stage does not exist, the command creates it.
     """
 
-    assert_project_type("streamlit")
+    pd = cli_context.project_definition
+    if not pd.meets_version_requirement("2"):
+        pd = _migrate_v1_streamlit_to_v2(pd)
 
-    streamlit: Streamlit = cli_context.project_definition.streamlit
-    if not streamlit:
-        return MessageResult("No streamlit were specified in project definition.")
+    streamlits: Dict[str, StreamlitEntity] = pd.get_entities_by_type(
+        entity_type="streamlit"
+    )
 
-    environment_file = streamlit.env_file
-    if environment_file and not Path(environment_file).exists():
-        raise ClickException(f"Provided file {environment_file} does not exist")
-    elif environment_file is None:
-        environment_file = "environment.yml"
+    if not streamlits:
+        raise NoProjectDefinitionError(
+            project_type="streamlit", project_file=cli_context.project_root
+        )
 
-    pages_dir = streamlit.pages_dir
-    if pages_dir and not Path(pages_dir).exists():
-        raise ClickException(f"Provided file {pages_dir} does not exist")
-    elif pages_dir is None:
-        pages_dir = "pages"
+    # TODO: fix in follow-up
+    if len(list(streamlits)) > 1:
+        raise ClickException(
+            "Currently only single streamlit entity per project is supported."
+        )
+
+    # Get first streamlit
+    streamlit: StreamlitEntity = streamlits[list(streamlits)[0]]
+
+    # Validate artefacts
+    for artefact in streamlit.artifacts:
+        if not artefact.exists():
+            raise ClickException(
+                f"Specified artefact {artefact} does not exist locally."
+            )
+
+    # Validate that main file is in artefacts
+    if Path(streamlit.main_file) not in streamlit.artifacts:
+        raise ClickException(
+            f"Specified main file {streamlit.main_file} is not included in artifacts."
+        )
```

Comment:
Maybe extract validation into separate function. WDYT?

Reply:
Good point, but I would prefer to do this in next PR where I will add support for multiple instances